plugin/v2/plugin.go

```diff
@@ -120,15 +125,37 @@ func (p *Plugin) InitPlugin(libRoot string) error {
 
 // Set is used to pass arguments to the plugin.
 func (p *Plugin) Set(args []string) error {
-	m := make(map[string]string, len(args))
-	for _, arg := range args {
-		i := strings.Index(arg, "=")
-		if i < 0 {
-			return fmt.Errorf("No equal sign '=' found in %s", arg)
+	p.Lock()
+	defer p.Unlock()
+
+	if p.PluginObj.Enabled {
+		return fmt.Errorf("cannot set on an active plugin, disable plugin before setting")
+	}
+
+	for _, s := range newSettables(args) {
+		// range over all the envs in the manifest
+		for _, env := range p.PluginObj.Manifest.Env {
```

Comment:
this won't work with https://github.com/docker/docker/pull/28086 since we are not only checking for env